Bug fixes

oskrkal opened the pull request:
This pull request contains fixes for two issues.

First, the tests in `tests/generation/code_test.py` didn't work on Windows. I updated the test, so it should be OS independent now.

Second, the module generating code from XSD sometimes generated class definition like the one below, which could not be compiled:
```
class MyType(__name__ + '.MyOtherType'):
    # ...
    pass
```
This was happening when one type was extending or restricting another type in XML schema and the base type name started with lower-case letter. I have extended the {{extension.xsd}} testing schema to demonstrate the issue.

Feel free to reach to me if you have any questions.

FelixSchwarz commented:
Hey, thank you very much for your contribution. The tests look quite ok, though I need some time to check. 

Ideally I would like to see more stand-alone tests instead of adding new stuff to the tests/assets/generation/extension.xsd (which makes reviewing+refactoring harder) but I'm glad for every contribution anyway.

Unless one of the other committers beat me to it I'll likely have time early next week to check. Keep up the good work :-)

oskrkal commented:
Thanks for the comments.

I have moved the changes in extension.xsd to a separate file and added new test method to xsd_code_test.py. I amended the changes to the original commit.

I also discovered that Jinja templates for code generation are not included in the distribution package, so I fixed that including a little mistake in the lib template.

oskrkal commented:
@FelixSchwarz, @pope1ni: Can one of you please look at this pull requested. It's been open for more than two months now.

pope1ni commented:
@oskrkal: Sorry for missing this. Could you reopen it against [soapteam/soapfish](/soapteam/soapfish) instead please and then I'll give it a review? TIA.

oskrkal commented:
@pope1ni: Created as https://github.com/soapteam/soapfish/pull/100. We can abandon this one, I guess.

pope1ni commented:
Thanks.
